tests/common/utils.py

```diff
@@ -66,3 +69,22 @@ def inverted_test(*arguments, **kwargs):
 
 
 fails = fails_with(AssertionError)
+
+
+def checks_deprecated_behaviour(func):
```

Comment:
Any suggestions?  I couldn't find any test that checked for raising a deprecation warning, so I don't think it's a drop-in replacement for something we're already doing.